### ISSUE TITLE
 Metric and visualization bugs with visualize_and_compare_testing_models.py

### ISSUE THREAD

**Issue by mariehbourget** [ivadomed/ivadomed#1138]
## Issue description
There was an error first reported here https://github.com/ivadomed/ivadomed/issues/1135#issuecomment-1142225824 when running the [visualize_and_compare_testing_models.py](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/scripts/visualize_and_compare_testing_models.py) script.
 
 ### Current behavior
There are 2 errors:
1. The script runs without the `--metric` flag, which outputs `dice_class0` by default.
However, it fails with the `--metric` flag (even when using `--metric dice_class0`).
So there seems to be something wrong with the argument parsing.

2. They also reported that the scripts runs without the `--metric` flag but the visualization still fails [here](https://github.com/ivadomed/ivadomed/issues/1135#issuecomment-1143731490).

I'm not very familiar with this script and could use some help to figure it out.
Maybe @mpompolas you have some insights?

**Comment by jcohenadad** [comment]
Would it make sense to instead do this visualization with WanDB, which has recently be made compatible with ivadomed? @naga-karthik @kiristern @Nish228

**Comment by kanishk16** [comment]
> There are 2 errors:
> 
> 1. The script runs without the `--metric` flag, which outputs `dice_class0` by default.
>    However, it fails with the `--metric` flag (even when using `--metric dice_class0`).
>    So there seems to be something wrong with the argument parsing.
I second this... I could debug this 

* Without `--metric` flag: 
![image](https://user-images.githubusercontent.com/36276423/172006065-70e5706d-da6c-47ff-9c2b-6b3605584009.png)

* with `--metric` flag (_raises an error_)
![image](https://user-images.githubusercontent.com/36276423/172006265-62dc6552-107c-492d-baa8-7088909ea2bf.png)


> 2. They also reported that the scripts runs without the `--metric` flag but the visualization still fails [here](https://github.com/ivadomed/ivadomed/issues/1135#issuecomment-1143731490).

I guess the visualization failure seems to be specific to Google Colab bcz I couldn't reproduce it on my machine. 

I'll open a PR shortly.

**Comment by kanishk16** [comment]
> Would it make sense to instead do this visualization with WanDB, which has recently be made compatible with ivadomed? @naga-karthik @kiristern @Nish228

Absolutely! So much so that Yang had created an issue for the same (#1088) but it was on a slightly lower priority as compared to other issues.
